src/IAPWS_WaterProperties.f90

```diff
@@ -3219,8 +3219,6 @@ SUBROUTINE regsopt(P, T, ireg)
   REAL(SRK), SAVE :: told = 0.0_SRK, pold = 0.0_SRK
   INTEGER(SIK), SAVE :: iregold = 0
 
-  IF (ireg /= 0 .AND. ireg /= 1 .AND. ireg /= 2 .AND. ireg /= 3 .AND. ireg /= 5 .AND. ireg /= 9) ireg = 0
```

Comment:
Should we leave an IF(PRESENT(ireg)) do this check.

Reply:
nevermind... its not optional

Reply:
maybe make ireg intent out only?

Reply:
Dang beat me to it. It appears to me that `ireg` is intent(out) in all cases. If possible it might be worth while to update that and then set it to 0 on entry.

Reply:
Yeah, good suggestions.  I made it intent out.  I also moved the `ireg=0` assignment up some, though not quite to the top.  I figured there was no reason to do 2 assignments in a row if t and p haven't changed.  I doubt a performance hit would ever be detectable but I think it's still pretty clear what's going on with the current version.

Reply:
Looks totally reasonable to me.